keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifact/ArtifactListener.kt

```diff
@@ -37,26 +38,25 @@ class ArtifactListener(
     event
       .artifacts
       .filter { it.type.toUpperCase() in artifactTypeNames }
-      .forEach { korkArtifact ->
-        if (artifactRepository.isRegistered(korkArtifact.name, korkArtifact.type())) {
-          val artifact = artifactRepository.get(korkArtifact.name, korkArtifact.type())
+      .forEach { artifact ->
+        if (artifactRepository.isRegistered(artifact.name, artifact.type())) {
           val version: String
           var status: ArtifactStatus? = null
-          when (artifact) {
-            is DebianArtifact -> {
-              version = "${korkArtifact.name}-${korkArtifact.version}"
-              status = debStatus(korkArtifact)
+          when (artifact.type()) {
+            DEB -> {
+              version = "${artifact.name}-${artifact.version}"
+              status = debStatus(artifact)
             }
-            is DockerArtifact -> {
-              version = korkArtifact.version
+            DOCKER -> {
+              version = artifact.version
             }
-            else -> throw UnsupportedArtifactTypeException(korkArtifact.type)
+            else -> throw UnsupportedArtifactTypeException(artifact.type)
           }
           log.info("Registering version {} ({}) of {} {}", version, status, artifact.name, artifact.type)
-          artifactRepository.store(artifact, version, status)
+          artifactRepository.store(artifact.name, artifact.type(), version, status)
             .also { wasAdded ->
               if (wasAdded) {
-                publisher.publishEvent(ArtifactVersionUpdated(artifact.name, artifact.type))
+                publisher.publishEvent(ArtifactVersionUpdated(artifact.name, artifact.type()))
```

Comment:
I don't understand why this has changed from a property to a function call. It seems to still be referenced as a property elsewhere

Reply:
This object (`artifact`) used to be a `DeliveryArtifact`, and that has the correct type as a property. Now, it's a `KorkArtifact`, and we need to parse the correct `ArtifactType` using the function call, because a `KorkArtifact` also has a `type` field on it, which is text.

Reply:
So that `type()` function is defined in Kork?

Reply:
No it's defined in `ArtifactListener`: https://github.com/spinnaker/keel/blob/master/keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifact/ArtifactListener.kt#L174